src/metabase/feature_extraction/values.clj

```diff
@@ -0,0 +1,45 @@
+(ns metabase.feature-extraction.values
+  "Helpers to get data behind various models in a consistent shape."
+  (:require [metabase.db.metadata-queries :as metadata]
+            [metabase.query-processor :as qp]))
+
+(defn field-values
+  "Return all the values of FIELD for QUERY."
+  [field query]
+  (->> (qp/process-query
```

Comment:
Any reason not to use the functions in `metabase.query-processor.middleware.expand` for building this?

Reply:
No (good) reason. Fixed. Thanks.